net-tools: rename from nettools

zimbatm opened the pull request:
By being in sync with the pname, it makes it easier to walk back from
the pname that shows in the /nix/store back to the attribute.

This change should not cause any rebuild.


<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [Nixpkgs 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/doc/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/doc/manual/release-notes/rl-2505.section.md) Nixpkgs Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
- [NixOS 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) NixOS Release notes)
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md), [pkgs/README.md](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md), [maintainers/README.md](https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md) and other contributing documentation in corresponding paths.

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

zimbatm committed:
net-tools: rename from nettools

By being in sync with the pname, it makes it easier to walk back from
the pname that shows in the /nix/store back to the attribute.

This change should not cause any rebuild.

maralorn commented:
Sadly your commit touches the autogenerated hackage-packages.nix file.

Annoyingly changing this would require modifying this line: https://github.com/NixOS/cabal2nix/blob/0caafdd58fa45228b2abc67078f239c85df21b89/cabal2nix/src/Distribution/Nixpkgs/Haskell/FromCabal/PostProcess.hs#L140

zimbatm commented on pkgs/development/libraries/gnutls/default.nix:
```diff
@@ -163,7 +163,7 @@ stdenv.mkDerivation rec {
     ]
     ++ lib.optionals doCheck [
       which
-      nettools
```
@maralorn it looks like liquid-fixpoint is already marked as broken in https://github.com/zimbatm/nixpkgs/blob/4442a92514a1ae88e81d5969f4b43298735f0404/pkgs/development/haskell-modules/configuration-hackage2nix/broken.yaml#L3646

Are those still being generated and pushed to nixpkgs?

zimbatm replied:
Ideally we can remove the package if it's broken, so I don't have to worry about cabal2nix backward/future-compatibility.

sternenseemann replied:
Nothing prevents you from bumping cabal2nix in this PR in principle.

zimbatm replied:
Done. Can we consider this point as "solved"?

zimbatm replied:
Done

6543 reviewed (commented):
I'm neither for nor against it ... so for my module just lgtm/rejeckt on what the majority of maintainers of this rename think.

as for my two cents: please dont bikeshed :) -  it it's good it is else well let it be ?!?

zimbatm committed:
haskellPackages.cabal2nix-unstable: 2025-04-30 -> 2025-06-14

Includes the nettool patch

zimbatm committed:
haskellPackages: regenerate package set based on current config (#416056)

This commit has been generated by maintainers/scripts/haskell/regenerate-hackage-packages.sh --fast

wolfgangwalther reviewed (approved):
LGTM

sternenseemann commented:
Should go to staging with this rebuild since the change is not particularly urgent.
